webserver/views/api/v1/datasets.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import absolute_import
+from flask import Blueprint, jsonify, request
+from flask_login import current_user
+from webserver.decorators import auth_required
+from webserver.views.api import exceptions as api_exceptions
+import db.dataset
+import db.exceptions
+
+bp_datasets = Blueprint('api_v1_datasets', __name__)
+
+
+@bp_datasets.route("/<uuid:dataset_id>", methods=["GET"])
+def get_dataset(dataset_id):
+    """Retrieve a dataset.
+
+    :resheader Content-Type: *application/json*
+    """
+    return jsonify(get_check_dataset(dataset_id))
+
+
+@bp_datasets.route("/", methods=["POST"])
+@auth_required
+def create_dataset():
+    """Create a new dataset.
+
+    **Example request**:
+
+    .. sourcecode:: json
+
+        {
+            "name": "Mood",
+            "description": "Dataset for mood classification.",
+            "public": true,
+            "classes": [
+                {
+                    "name": "Happy",
+                    "description": "Recordings that represent happiness.",
+                    "recordings": ["770cc467-8dde-4d22-bc4c-a42f91e"]
+                },
+                {
+                    "name": "Sad"
+                }
+            ]
+        }
+
+    :reqheader Content-Type: *application/json*
+    :<json string name: *Required.* Name of the dataset.
+    :<json string description: *Optional.* Description of the dataset.
+    :<json boolean public: *Optional.* ``True`` to make dataset public, ``false`` to make it private. New datasets are
+        public by default.
+    :<json array classes: *Optional.* Array of objects containing information about classes to add into new dataset. For
```

Comment:
Submitting json with no `public` or `classes` results in errors thrown on the server:

    ValidationException: Field `public` is missing from dataset dictionary.